### PR TITLE
Salesforce-asim-authentication parser

### PR DESCRIPTION
**Pull request by rushriva**
Required items, please complete
   
   Change(s):
   - Added asim authentication parser for Salesforce

   Reason for Change(s):
   - Salesforce content enrichment

   Version Updated:
   - NA
   

   Testing Completed:
   - yes

   Checked that the validations are passing and have addressed any issues that are present:
   - NA

# Guidance <- remove section before submitting
-----------------------------------------------------------------------------------------------------------

### CODE REVIEW

**Comment by oshezaf** [comment]
Hi @rushriva: to help me review - 
- Where are the sample logs located?
- Did you run the ASIM testers? If so, can you share the restults?

File: Parsers/ASimAuthentication/Parsers/ASimAuthenticationSalesforceSC.yaml
Comment (oshezaf) [inline Parsers/ASimAuthentication/Parsers/ASimAuthenticationSalesforceSC.yaml:46]: Do we know what is user_id_derived_s?

File: Parsers/ASimAuthentication/Parsers/ASimAuthenticationSalesforceSC.yaml
Comment (oshezaf) [inline Parsers/ASimAuthentication/Parsers/ASimAuthenticationSalesforceSC.yaml:47]: Do we know what's the difference from user_email_s?

File: Parsers/ASimAuthentication/Parsers/ASimAuthenticationSalesforceSC.yaml
Comment (oshezaf) [inline Parsers/ASimAuthentication/Parsers/ASimAuthenticationSalesforceSC.yaml:52]: browser_type is always empty in the sample set. Do we know, from documentation maybe, that it has a relevant value?

**Review by oshezaf — DISMISSED** [review]
First, while there are many comments... You did a very good job. Don't let the many comments here deter you.

As to the comments: 
- You need to create the vim parser as well. Can be a second step after this one is finished.
- Did you add sample data based on the workspace to either the public repository or the private one (Prateek can help with the datails)?
- You probably did not run the data tester.
- I noticed that you added the project statement after testing, which is good. Notice a missing field (see later).
- I need to update testing. There are fields that need setting and are not flagged. Will do it ASAP. It might lead to additional notes.
- Additional mappings
  - api_version_s -> EventProductVersion
  - organization_id_s -> TargetUserScope (this is a very new additional to the schema, needed for UEBA)
  - cipher_suite_s -> TlsCipher (not yet defined, but seems useful, will be added)
  - tls_protocol_s -> TlsVersion (not yet defined, but seems useful, will be added)
  - Do you know what login_key_s is?

File: Parsers/ASimAuthentication/Parsers/ASimAuthenticationSalesforceSC.yaml
Comment (rushriva) [reply Parsers/ASimAuthentication/Parsers/ASimAuthenticationSalesforceSC.yaml:39]: Changed case to iff

File: Parsers/ASimAuthentication/Parsers/ASimAuthenticationSalesforceSC.yaml
Comment (rushriva) [reply Parsers/ASimAuthentication/Parsers/ASimAuthenticationSalesforceSC.yaml:41]: SaleforceId is added

File: Parsers/ASimAuthentication/Parsers/ASimAuthenticationSalesforceSC.yaml
Comment (rushriva) [reply Parsers/ASimAuthentication/Parsers/ASimAuthenticationSalesforceSC.yaml:53]: Project-reorder removed

File: Parsers/ASimAuthentication/Parsers/ASimAuthenticationSalesforceSC.yaml
Comment (rushriva) [reply Parsers/ASimAuthentication/Parsers/ASimAuthenticationSalesforceSC.yaml:49]: request_id_s is projected

File: Parsers/ASimAuthentication/Parsers/ASimAuthenticationSalesforceSC.yaml
Comment (rushriva) [reply Parsers/ASimAuthentication/Parsers/ASimAuthenticationSalesforceSC.yaml:44]: Completed

File: Parsers/ASimAuthentication/Parsers/ASimAuthenticationSalesforceSC.yaml
Comment (rushriva) [reply Parsers/ASimAuthentication/Parsers/ASimAuthenticationSalesforceSC.yaml:72]: Update done

File: Parsers/ASimAuthentication/Parsers/ASimAuthenticationSalesforceSC.yaml
Comment (rushriva) [reply Parsers/ASimAuthentication/Parsers/ASimAuthenticationSalesforceSC.yaml:48]: @oshezaf  - Salesforce document link - https://developer.salesforce.com/docs/atlas.en-us.object_reference.meta/object_reference/sforce_api_objects_eventlogfile_login.htm
User type in salesforce is the category of user license. e.g. CsnOnly, CspLitePortal, CustomerSuccess, guest, PowerCustomerSuccess, PowerPartner, SelfService, Standard

File: Parsers/ASimAuthentication/Parsers/ASimAuthenticationSalesforceSC.yaml
Comment (rushriva) [reply Parsers/ASimAuthentication/Parsers/ASimAuthenticationSalesforceSC.yaml:37]: Updated as per the comment

File: Parsers/ASimAuthentication/Parsers/ASimAuthenticationSalesforceSC.yaml
Comment (rushriva) [reply Parsers/ASimAuthentication/Parsers/ASimAuthenticationSalesforceSC.yaml:46]: @oshezaf  - user_id_derived_s - The 18-character case insensitive ID of the user who’s using Salesforce services through the UI or the API.

File: Parsers/ASimAuthentication/Parsers/ASimAuthenticationSalesforceSC.yaml
Comment (rushriva) [reply Parsers/ASimAuthentication/Parsers/ASimAuthenticationSalesforceSC.yaml:47]: user_name_s and user_email_s have same value

File: Parsers/ASimAuthentication/Parsers/ASimAuthenticationSalesforceSC.yaml
Comment (rushriva) [reply Parsers/ASimAuthentication/Parsers/ASimAuthenticationSalesforceSC.yaml:52]: browser_type - 
The identifier string returned by the browser used at login.
Example values are:
Go-http-client/1.1
Mozilla/5.0 (Macintosh; Intel Mac OS X 10.12; rv%3A50.0) Gecko/20100101 Firefox/50.0
Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/51.0.2704.84 Safari/537.36

**Comment by rushriva** [comment]
> First, while there are many comments... You did a very good job. Don't let the many comments here deter you.
> 
> As to the comments:
> 
> * You need to create the vim parser as well. Can be a second step after this one is finished.
> * Did you add sample data based on the workspace to either the public repository or the private one (Prateek can help with the datails)?
> * You probably did not run the data tester.
> * I noticed that you added the project statement after testing, which is good. Notice a missing field (see later).
> * I need to update testing. There are fields that need setting and are not flagged. Will do it ASAP. It might lead to additional notes.
> * Additional mappings
>   
>   * api_version_s -> EventProductVersion
>   * organization_id_s -> TargetUserScope (this is a very new additional to the schema, needed for UEBA)
>   * cipher_suite_s -> TlsCipher (not yet defined, but seems useful, will be added)
>   * tls_protocol_s -> TlsVersion (not yet defined, but seems useful, will be added)
>   * Do you know what login_key_s is?

@oshezaf  - I will create vim parser once asim parser is approved with changes.

**Comment by v-prasadboke** [comment]
Hello @v-sabiraj please look into this

**Comment by v-prasadboke** [comment]
Hello @v-sabiraj any updates on the above

**Comment by v-prasadboke** [comment]
Hello @v-sabiraj waiting for your feedback

**Comment by v-prasadboke** [comment]
Hello @oshezaf please look into the changes

File: Parsers/ASimAuthentication/Parsers/ASimAuthenticationSalesforceSC.yaml
Comment (devikamehra) [reply Parsers/ASimAuthentication/Parsers/ASimAuthenticationSalesforceSC.yaml:174]: Mapping this to HttpUserAgent

File: Parsers/ASimAuthentication/Parsers/ASimAuthenticationSalesforceSC.yaml
Comment (devikamehra) [reply Parsers/ASimAuthentication/Parsers/ASimAuthenticationSalesforceSC.yaml:165]: Potential candidate for ActorUserId and ActorUserIdType = "SaleforceId"

File: Parsers/ASimAuthentication/Parsers/ASimAuthenticationSalesforceSC.yaml
Comment (devikamehra) [reply Parsers/ASimAuthentication/Parsers/ASimAuthenticationSalesforceSC.yaml]: This would require TargetUserType to be set to "Other"

File: Parsers/ASimAuthentication/Parsers/ASimAuthenticationSalesforceSC.yaml
Comment (vakohl) [reply Parsers/ASimAuthentication/Parsers/ASimAuthenticationSalesforceSC.yaml]: we have 3:
image

File: Parsers/ASimAuthentication/Parsers/ASimAuthenticationSalesforceSC.yaml
Comment (vakohl) [reply Parsers/ASimAuthentication/Parsers/ASimAuthenticationSalesforceSC.yaml]: not doing

File: Parsers/ASimAuthentication/Parsers/ASimAuthenticationSalesforceSC.yaml
Comment (vakohl) [reply Parsers/ASimAuthentication/Parsers/ASimAuthenticationSalesforceSC.yaml]: EventOriginalUid = request_id_s,

File: Parsers/ASimAuthentication/Parsers/ASimAuthenticationSalesforceSC.yaml
Comment (vakohl) [reply Parsers/ASimAuthentication/Parsers/ASimAuthenticationSalesforceSC.yaml]: image

File: Parsers/ASimAuthentication/Parsers/ASimAuthenticationSalesforceSC.yaml
Comment (vakohl) [reply Parsers/ASimAuthentication/Parsers/ASimAuthenticationSalesforceSC.yaml]: done:
image

File: Parsers/ASimAuthentication/Parsers/ASimAuthenticationSalesforceSC.yaml
Comment (vakohl) [reply Parsers/ASimAuthentication/Parsers/ASimAuthenticationSalesforceSC.yaml]: https://developer.salesforce.com/docs/atlas.en-us.object_reference.meta/object_reference/sforce_api_objects_eventlogfile_login.htm?q=Login

we are not using this value

File: Parsers/ASimAuthentication/Parsers/ASimAuthenticationSalesforceSC.yaml
Comment (vakohl) [reply Parsers/ASimAuthentication/Parsers/ASimAuthenticationSalesforceSC.yaml]: this is done

Comment (vakohl) [reply Parsers/ASimAuthentication/Parsers/ASimAuthenticationSalesforceSC.yaml]: @devikamehra sample logs: https://github.com/Azure/Azure-Sentinel/blob/master/Sample%20Data/Custom/SalesforceServiceCloud_CL.json

**Comment by vakohl** [comment]
@v-atulyadav can you please check the one validation which shows Queued?

File: Parsers/ASimAuthentication/Parsers/ASimAuthenticationSalesforceSC.yaml
Comment (anki-narravula) [inline Parsers/ASimAuthentication/Parsers/ASimAuthenticationSalesforceSC.yaml:27]: These fields are missing - request_status_s, wave_session_id_g
,

File: Parsers/ASimAuthentication/Parsers/ASimAuthenticationSalesforceSC.yaml
Comment (anki-narravula) [inline Parsers/ASimAuthentication/Parsers/ASimAuthenticationSalesforceSC.yaml:338]: I think _ (underscore fields are hidden and not require to project away).

File: Parsers/ASimAuthentication/Parsers/vimAuthenticationSalesforceSC.yaml
Comment (anki-narravula) [inline Parsers/ASimAuthentication/Parsers/vimAuthenticationSalesforceSC.yaml:295]: computing it several times. Performance wise, is this better than using extended and run the filter on it ?

File: Parsers/ASimAuthentication/Tests/Salesforce_ASimAuthentication_DataTest.csv
Comment (anki-narravula) [inline Parsers/ASimAuthentication/Tests/Salesforce_ASimAuthentication_DataTest.csv:3]: Can we replace Salesforce.com IP with some internal loopback IP 127.0.0.1 or 0.0.0.0

File: Parsers/ASimAuthentication/Tests/Salesforce_ASimAuthentication_DataTest.csv
Comment (anki-narravula) [inline Parsers/ASimAuthentication/Tests/Salesforce_ASimAuthentication_DataTest.csv:5]: Please add this value to Test case

File: Parsers/ASimAuthentication/Tests/Salesforce_ASimAuthentication_DataTest.csv
Comment (anki-narravula) [inline Parsers/ASimAuthentication/Tests/Salesforce_ASimAuthentication_DataTest.csv:6]: Please include in test case

File: Parsers/ASimAuthentication/Tests/Salesforce_ASimAuthentication_DataTest.csv
Comment (anki-narravula) [inline Parsers/ASimAuthentication/Tests/Salesforce_ASimAuthentication_DataTest.csv:2]: Please add this value to the test case and documentation

File: Parsers/ASimAuthentication/Parsers/ASimAuthenticationSalesforceSC.yaml
Comment (vakohl) [reply Parsers/ASimAuthentication/Parsers/ASimAuthenticationSalesforceSC.yaml:277]: done

File: Parsers/ASimAuthentication/Parsers/ASimAuthenticationSalesforceSC.yaml
Comment (vakohl) [reply Parsers/ASimAuthentication/Parsers/ASimAuthenticationSalesforceSC.yaml:27]: done

File: Parsers/ASimAuthentication/Parsers/ASimAuthenticationSalesforceSC.yaml
Comment (vakohl) [reply Parsers/ASimAuthentication/Parsers/ASimAuthenticationSalesforceSC.yaml:338]: @anki-narravula removing _resourceID, keeping _ItemId. _ItemId is visible if not done project-away

File: Parsers/ASimAuthentication/Tests/Salesforce_ASimAuthentication_DataTest.csv
Comment (vakohl) [reply Parsers/ASimAuthentication/Tests/Salesforce_ASimAuthentication_DataTest.csv:3]: 'Salesforce.com IP' is the IPAddress value in sample data. thats what coming in this error. This is not an actual error. We have actual IP Address also coming in sample data.
image

File: Parsers/ASimAuthentication/Tests/Salesforce_ASimAuthentication_DataTest.csv
Comment (vakohl) [reply Parsers/ASimAuthentication/Tests/Salesforce_ASimAuthentication_DataTest.csv:5]: @anki-narravula I have changed EventProduct to 'Service Cloud'
EventVendor: Salesforce
EventProduct: Service Cloud
are already added in tester.csv

File: Parsers/ASimAuthentication/Tests/Salesforce_ASimAuthentication_DataTest.csv
Comment (vakohl) [reply Parsers/ASimAuthentication/Tests/Salesforce_ASimAuthentication_DataTest.csv:2]: @anki-narravula has been added already in tester.csv
image

**Comment by vakohl** [comment]
@microsoft-github-policy-service agree [company="Microsoft"]

**Comment by vakohl** [comment]
> @rushriva please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @microsoft-github-policy-service agree [company="{your company}"]
> ```
> 
> > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree company="Microsoft"
> > ```
> 
> Contributor License Agreement

@microsoft-github-policy-service agree [company="Microsoft"]

**Comment by vakohl** [comment]
> @rushriva please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @microsoft-github-policy-service agree [company="{your company}"]
> ```
> 
> > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree company="Microsoft"
> > ```
> 
> Contributor License Agreement

@microsoft-github-policy-service agree company="Microsoft"

**Comment by vakohl** [comment]
> @rushriva please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @microsoft-github-policy-service agree [company="{your company}"]
> ```
> 
> > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree company="Microsoft"
> > ```
> 
> Contributor License Agreement

@microsoft-github-policy-service agree [company="Microsoft"]

@microsoft-github-policy-service agree

@microsoft-github-policy-service agree company="Microsoft"

**Comment by vakohl** [comment]
@anki-narravula can you please validate the comments?

**Review by anki-narravula — DISMISSED** [review]

**Comment by v-sudkharat** [comment]
Hi @vakohl, This branch has conflicts. Could you please check and resolve it. Thanks!

**Review by anki-narravula — APPROVED** [review]

**Comment by v-sudkharat** [comment]
@microsoft-github-policy-service agree [company="Microsoft"]

**Review by v-atulyadav — APPROVED** [review]

**Pull request merged**
